server/channels/store/storetest/job_store.go

```diff
@@ -56,6 +59,51 @@ func testJobSaveGet(t *testing.T, ss store.Store) {
 	require.Equal(t, "12345", received.Data["Total"])
 }
 
+func testJobSaveOnce(t *testing.T, ss store.Store) {
+	var wg sync.WaitGroup
+
+	ids := make([]string, 2)
+	for i := 0; i < 2; i++ {
+		wg.Add(1)
+		go func(){
+			defer wg.Done()
+			job := &model.Job{
+				Id:     model.NewId(),
+				Type:   model.JobTypeS3PathMigration,
+				Status: model.JobStatusPending,
+				Data: map[string]string{
+					"Processed":     "0",
+					"Total":         "12345",
+					"LastProcessed": "abcd",
+				},
+			}
+
+			job, err := ss.Job().SaveOnce(job)
+			if err != nil {
+				var pqErr *pq.Error
+				if errors.As(err, &pqErr) {
+					t.Logf("%#v\n", pqErr)
+				}
+			}
+			require.NoError(t, err)
+
+			if job != nil {
+				ids[0] = job.Id
```

Comment:
nit: this should be `i` AFAIU.
```suggestion
				ids[i] = job.Id
```

Reply:
lol .. right